protocol/xmrtaker/message_handler.go

```diff
@@ -148,6 +143,52 @@ func (s *swapState) handleSendKeysMessage(msg *message.SendKeysMessage) (common.
 	return out, nil
 }
 
+func (s *swapState) checkForXMRLock() {
+	// monero block time is >1 minute, so this should be fine
+	const checkForXMRLockInterval = time.Minute
```

Comment:
It might be nice if we could lower this for testing.  We could have a `BlockInterval` variable defined to 2 minutes in the `monero` package.  Then in `TestBackgroundMineBlocks`, it could get lowered to 1 second.  Then this constant here could be half the block interval.  (Just throwing out ideas ... you may have a better one.) What I described above, is similar to what we do with the `blockSleepDuration` variable.

Another idea: You could call `WaitForBlocks` passing 1 for the number of blocks. That method handles the context being cancelled.

Reply:
I updated this function to just set the interval to 1s if we're on development. I decided against `WaitForBlocks` since it returns errors in other cases than just context cancel which isn't necessarily what we want here